lib/Dialect/HW/HWOps.cpp

```diff
@@ -1469,6 +1469,13 @@ void StructExtractOp::build(OpBuilder &builder, OperationState &odsState,
   build(builder, odsState, field.type, input, field.name);
 }
 
+void StructExtractOp::build(OpBuilder &builder, OperationState &odsState,
+                            Value input, StringRef field) {
```

Comment:
This makes a lot of sense, but please change this helper to take a StringAttr instead of a StringRef.  It looks like StructAttr is currently defined wrong (encouraging stringref) but we [should fix that](https://github.com/llvm/circt/issues/2324) :-)

Reply:
I changed. Thanks! Also the issue will be fixed in https://github.com/llvm/circt/pull/2325